### DISCUSSION TITLE
Is it a way for REALITY servers to be undetected even with 20-30 clients?

### DISCUSSION THREAD

**Discussion by lostsoul6** [XTLS/Xray-core#2001] [Q&A]
Hello Guys ,

I'm using the new REALITY option on a server . Since this method is direct , I think it would be really easy for GFW to detect the servers with 20-30 clients because of the number of connections and the volume of data .

I'm not an IT tech so my following questions might sound stupic but I was wondering if you could answer :

1 - Since REALITY only supports TCP , is it a way we can hide the IP address of the server behind a CDN ? Is there any way that the IP of our server is hidden and not detected by GFW ?

2 - Will there be support for websocket and grpc added later so we can use CDN to hide the IP ?

3 - I have 5-6 clients on one server and it's not yet blocked , do you think it would be detected if it was 20-30 clients or should it be more clients like 50 or more for it to be detected ?

4 - A few months ago before the recent limitations in Iran , I was using Caddy with VMESS + WS . caddy received the requests on port 443 and passed the real ones to Xray and the fake requests were redirected to a fake site . Can we have such mechanism with REALITY protocol ?

Regards ,

**Comment by us254** [comment]
REALITY has gRPC support already.but even with this reality not support proxy by CDN.

in Iran, they block most of cloudflare  servers, so you can't use Cloudflare CDN. You have to use your own server address, which is more stable and less likely to be blocked. If you find a server from Cloudflare that is not blocked, it will be blocked soon anyway. This is not because they know you are using it, but maybe because other people are using it in a bad way or maybe because they block Cloudflare servers randomly.

**Reply by lostsoul6** [reply]
> REALITY has gRPC support already.but even with this reality not support proxy by CDN.
> 
> in Iran, they block most of cloudflare servers, so you can't use Cloudflare CDN. You have to use your own server address, which is more stable and less likely to be blocked. If you find a server from Cloudflare that is not blocked, it will be blocked soon anyway. This is not because they know you are using it, but maybe because other people are using it in a bad way or maybe because they block Cloudflare servers randomly.

if I use my own server address , there is no protection on server IP and they can block it easily . Have you checked it with 20-30 clients on a single server to see if it would survive ?
